hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OmSnapshotUtils.java

```diff
@@ -137,13 +139,31 @@ public static void createHardLinks(Path dbPath) throws IOException {
     }
   }
 
+  private static void createHardLinksOrCopy(Path fullFromPath, Path fullToPath) throws IOException {
+    if (isSamePartition(fullFromPath.getParent(), fullToPath.getParent())) {
+      Files.createLink(fullToPath, fullFromPath);
+    } else {
+      Files.copy(fullFromPath, fullToPath, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Is replace existing correct? Can this kind of a situation occur where we are re-copying stuff? We shouldn't inadvertently delete data.

Reply:
Can we do some kind of a checksum verification if the fullToPath exists. 
1) copy to tmp file 
2) Verify checksum if the files exist
3) if checksums don't match raise exception.

Reply:
The file shouldn't exist at all, I'm doing a replace just as a precaution. I'm following the normal OM bootstrap process, We don't do a checksum check for that either.